### ISSUE TITLE
Bug: Struct comparison

### ISSUE THREAD

**Issue by acquamarin** [kuzudb/kuzu#4545]
### Kùzu version

master

### What operating system are you using?

MAC M1

### What happened?

If two structs has different field name, duckdb treats them as two different structs:
```
select {c: 2, d: 4} =  {c: 2, e: 4} ;
```
Raises:
```
Mismatch Type Error: Type STRUCT(c INTEGER, e INTEGER) does not match with STRUCT(c INTEGER, d INTEGER). Cannot cast STRUCTs - element "e" in source struct was not found in target struct
LINE 1: select {c: 2, d: 4} =  {c: 2, e: 4} 
```
However, Kuzu returns true in such case. 
I think it is a good idea to be consistent with duckdb's behaviour.

**Comment by SterlingT3485** [comment]
may related to #4155 , looking into it

**Comment by SterlingT3485** [comment]
This will be also fixed by #4560
